packages/salesforcedx-vscode-core/src/commands/forceCreateManifest.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { ComponentSet } from '@salesforce/source-deploy-retrieve';
+import * as fs from 'fs';
+import { join } from 'path';
+import { format } from 'util';
+import * as vscode from 'vscode';
+import { nls } from '../messages';
+import { getRootWorkspacePath } from '../util';
+
+export async function forceCreateManifest(
+  sourceUri: vscode.Uri,
+  uris: vscode.Uri[] | undefined
+) {
+  if (!uris || uris.length < 1) {
+    uris = [];
+    uris.push(sourceUri);
+  }
+  const sourcePaths = uris.map(uri => uri.fsPath);
+  if (sourcePaths) {
+    const componentSet = ComponentSet.fromSource(sourcePaths);
+    const inputOptions = {
+        placeHolder: nls.localize('manifest_editor_save_placeholder'),
+        prompt: nls.localize('manifest_editor_save_prompt'),
+        validateInput: text => {
+          return text.includes('.xml') ? 'Remove file extension' : null;
+        }
+    } as vscode.InputBoxOptions;
+    const response = await vscode.window.showInputBox(inputOptions);
+    if (response === undefined) {
```

Comment:
any reason you prefer the specific `===` to just `if (!response)`?

Reply:
This pattern is present elsewhere, and undefined has a specific meaning to signify the escape key, so wanted to be clear on that. https://code.visualstudio.com/api/references/vscode-api#window

Reply:
word. That makes sense.  empty string would get through (!response) so the strict equality is def better